inventory-provisioning-web/src/main/scala/com/normation/inventory/provisioning/endpoint/FusionReportEndpoint.scala

```diff
@@ -94,64 +97,131 @@ class FusionReportEndpoint(
+                    val fail = eb ?~! "Error when trying to check inventory key status"
+                    logger.error(fail.messageChain)
+                    logger.debug(s"Time to error: ${printer.print(new Duration(start, System.currentTimeMillis).toPeriod)}")
+                    new ResponseEntity(fail.messageChain, HttpStatus.PRECONDITION_FAILED)
                 }
-              case f@Failure(_,_,_) =>
-                val msg = "Error when trying to parse inventory: %s".format(f.failureChain.map( _.msg).mkString("\n", "\ncause: ", "\n"))
-                logger.error(msg)
-                f.rootExceptionCause.foreach { exp => logger.error("Exception was: ", exp) }
-                logger.debug("Time to error: %s".format(printer.print(new Duration(start, System.currentTimeMillis).toPeriod)))
-                new ResponseEntity(msg, HttpStatus.PRECONDITION_FAILED)
-              case _ =>
-                val msg = "The inventory is empty, not saving anything."
-                logger.error(msg)
-                logger.debug("Time to error: %s".format(printer.print(new Duration(start, System.currentTimeMillis).toPeriod)))
-                new ResponseEntity(msg, HttpStatus.PRECONDITION_FAILED)
             }
-          } catch {
-            case e:Exception =>
-              val msg = "Exception when processing inventory '%s'".format(reportFile.getOriginalFilename)
-              logger.error(msg)
-              logger.error("Reported exception is: ", e)
-              new ResponseEntity(msg+"\n", HttpStatus.PRECONDITION_FAILED)
-          } finally {
-            in.close
-          }
-        } else {
-          //more than one file: we don't know the one to take, so we just ask the user to only send one file
-          defaultBadAnswer("Too many files")
+
+          // Error during parsing
+          case eb : EmptyBox =>
+            val fail = eb ?~! "Error when trying to parse inventory"
+            logger.error(fail.messageChain)
+            fail.rootExceptionCause.foreach { exp => logger.error(s"Exception was: ${exp}") }
+            logger.debug(s"Time to error: ${printer.print(new Duration(start, System.currentTimeMillis).toPeriod)}")
+            new ResponseEntity(fail.messageChain, HttpStatus.PRECONDITION_FAILED)
         }
+      }
+    }
 
-      case _ =>
-        defaultBadAnswer("No inventory")
+    request match {
+      case multipart: DefaultMultipartHttpServletRequest =>
+        import scala.collection.JavaConversions._
+        val params : Map[String,MultipartFile]= multipart.getFileMap.toMap
+
+        /*
+         * params are :
+         * * 'file' => The inventory file
+         * * 'signature' => The signature file
+         */
+        val inventoryParam = "file"
+        val signatureParam = "signature"
+
+        (params.get(inventoryParam),params.get(signatureParam)) match {
+          // No inventory, error
+          case (None,_) =>
+            defaultBadAnswer("No inventory")
```

Comment:
I thing that in the message should be modified to say that the param name MUST be file (before, it was not important): 

 def defaultBadAnswer(reason: String) = new ResponseEntity(s"""${reason} sent. You have to POST a request with exactly one file in attachment (with 'content-disposition': file), for parameter name "file"
                               |For example, for curl, use: curl -F "file=@path/to/file"
                               |""".stripMargin, HttpStatus.PRECONDITION_FAILED)